### PR TITLE
feat: add form components

### PR DESCRIPTION
**Pull request by theodorusclarence**
# Description & Technical Solution

Form components, tested for:
- readonly
- disabled
- icons for input

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] Already rebased against main branch.

# Screenshots

https://user-images.githubusercontent.com/55318172/211809249-a1d7c6e4-0f4d-489e-b4d5-2c8236528ac4.mp4

### CODE REVIEW

**Comment by ppdbultimate** [comment]
#### 📝 Changed routes:


| [**/index.page**](https://design-system-moon-qgg22bn02-ppdbultimate.vercel.app/index.page) | [**/sandbox/form.page**](https://design-system-moon-qgg22bn02-ppdbultimate.vercel.app/sandbox/form.page) |
|:-:|:-:|
|<a href="https://design-system-moon-qgg22bn02-ppdbultimate.vercel.app/index.page">Screenshot of /index.page</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/kJI-4uMA2X6nroNmMReCU__J4WJX1zGdUYAtZjk04C2z5Cd34vmoDMmFTaDo1cvZnqDA1rjXvSuEUBPBnZwNlg.png">(view full size)</a> |<a href="https://design-system-moon-qgg22bn02-ppdbultimate.vercel.app/sandbox/form.page">Screenshot of /sandbox/form.page</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/trv_CeZe47MvFt-AvaAW-hsEKHbDK-N0Ijd0U6za9DtDL4FImaVOWfnOqFtwnG42waKgnIBpxarPius05D2W1A.png">(view full size)</a> |







Commit 7036853d8c9d53d044e3e59ec9477314a32ddf26 (https://design-system-moon-qgg22bn02-ppdbultimate.vercel.app).

File: src/components/forms/Input.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        <input
+          {...register(id, validation)}
+          {...rest}
+          type={type}
+          name={id}
+          id={id}
+          readOnly={readOnly}
+          disabled={disabled}
+          className={clsx(
+            'block w-full rounded-[0.25rem] shadow-sm',
```
Comment (winatungmiharja) [inline src/components/forms/Input.tsx:83]: kalau base 16px brrti ini rounded nya 4px, di design system itu rounded nya 8px (rounded-lg), mau disamain mas?

**Comment by winatungmiharja** [comment]
error state file preview punya ring lebih tebel, intend nyaa memang ka mas?
image

**Comment by winatungmiharja** [comment]
yang list optionnya sepertinya perlu disamakan rounded nya juga mas
image

**Comment by theodorusclarence** [comment]
image
CleanShot 2023-01-11 at 21 07 14

**Review by winatungmiharja — APPROVED** [review]

**Review by rizqitsani — APPROVED** [review]

**Pull request merged**